src/framework/cfgTools.ml

```diff
@@ -0,0 +1,540 @@
+open MyCFG
+open Cil
+open Pretty
+open GobConfig
+
+module H = NodeH
+module NH = NodeH
+
+
+let do_the_params (fd: fundec) =
+  (* This function used to create extra variables, but now it just sets the
+   * vdecl to -3, lovely... *)
+  let create_extra_var (p: varinfo): unit =
+    match p.vtype with
+    | TPtr (t,_) -> p.vdecl <- {p.vdecl with line = -3 }
+    | _ -> p.vdecl <- {p.vdecl with line = -3 }
+  in
+  List.iter create_extra_var fd.sformals
```

Comment:
In the long run, I would want to get rid of this setting locations to -3 to signal parameter. Maybe it would be nicer to have a hashtable of parameter `vid`s?

Also, is `line = -4` with meaning `Context` ever used anywhere.

Reply:
In the mean time, why does this distinguish by `vtype` and then does nothing with it?

Reply:
> Also, is `line = -4` with meaning `Context` ever used anywhere?

At some point I looked into it and also couldn't find it being used.

> In the long run, I would want to get rid of this setting locations to -3 to signal parameter. Maybe it would be nicer to have a hashtable of parameter `vid`s?

Indeed, these magic constants are spread around all over the place. No surprise that -4 has been removed from some places but still exists in others. And instead of a hashtable of `vid`s, it's much nicer to have a hashtable of `CilType.Varinfo` to let that take care of the identity automatically, not spreading the `vid` usage everywhere.

Actually, this could be related to `Cilfacade.find_scope_fundec`. Namely, if one generalizes that to not just store `fundec`s as values but variants like `Formal of fundec | Local of fundec | Global`, then the grouping info could be done using the same mapping that stores the scopes. But I'll leave the grouping cleanup for a separate PR.